src/gui/gtk.c

```diff
@@ -2682,20 +2683,21 @@ static void _ui_log_redraw_callback(gpointer instance,
 static void _ui_toast_redraw_callback(gpointer instance,
                                       GtkWidget *widget)
 {
+  dt_control_t *dc = darktable.control;
   // draw toast message, if any
-  dt_pthread_mutex_lock(&darktable.control->toast_mutex);
-  if(darktable.control->toast_ack != darktable.control->toast_pos)
+  dt_pthread_mutex_lock(&dc->toast_mutex);
+  if(dc->toast_ack != dc->toast_pos)
   {
-    const uint32_t first_message = MAX(darktable.control->toast_ack,
-                                       darktable.control->toast_pos - DT_CTL_TOAST_SIZE + 1);
+    const int numlines = MAX(0, MIN(DT_CTL_TOAST_SIZE - 1, dt_conf_get_int("message_list_length")));
+    const int32_t first_message = MAX(dc->toast_ack, dc->toast_pos - numlines);
```

Comment:
Have you verified that this works?  toast_pos and toast_ack are unsigned, so when numlines>toast_pos, I would expect first_message to be 4 billion....  Forcing numlines to be no more than toast_pos would definitely work, though.

Reply:
toast_pos and toast_ack are now int32_t